syntax = "proto3";

service StableDiffusionService {
  rpc TextToImage(stream ImagineRequest) returns (ImagineResponse) {};
}

message ImagineRequest {
  string prompt = 1;
  bool from_file = 2;
  optional uint32 ddim_steps = 3;
  optional uint32 n_iter = 4;
  optional uint32 H = 5;
  optional uint32 W = 6;
  optional uint32 C = 7;
  optional uint32 f = 8;
  optional uint32 n_samples = 9;
  optional uint32 n_rows = 10;
  optional int64 seed = 11;
  optional bool turbo = 12;
  optional uint32 scale = 13;
}

message ImagineResponse {
  repeated ImageData image_data = 1;
}

message ImageData {
  string seed = 1;
  bytes chunk_data = 2;
}